ReactiveCocoa/Swift/Signal.swift

```diff
@@ -410,16 +410,126 @@ public func takeUntil<T, E>(trigger: Signal<(), NoError>)(signal: Signal<T, E>)
 	}
 }
 
+/// Forwards events from `signal` with history: values of the returned signal
+/// are a tuple whose first member is the previous value and whose second member
+/// is the current value. `initial` is supplied as the first member when `signal`
+/// sends its first value.
+public func combinePrevious<T, E>(initial: T)(signal: Signal<T, E>) -> Signal<(T, T), E> {
+	return signal |> scan((initial, initial)) { previousCombinedValues, newValue in
+		return (previousCombinedValues.1, newValue)
+	}
+}
+
+/// Like `scan`, but sends only the final value and then immediately completes.
+/// If `signal` completes before you observe the resulting signal, the resulting
+/// signal may send no values before completing.
```

Comment:
Yep, that's right. It would just complete immediately. 

Do you want me to remove the comment, since it applies to all these operators? Or do you want to support `reduce` on only `SignalProducer`s?